File: Solutions/2251. Number of Flowers in Full Bloom/Python/Number_of_Flowers in_Full_Bloom.py
from typing import List 
import bisect

class Solution:
    def fullBloomFlowers(self, flowers: List[List[int]], people: List[int]) -> List[int]:
        start = sorted([s for s, e in flowers])
        end = sorted([e for s, e in flowers])
        return [bisect.bisect_right(start, t) - bisect.bisect_left(end, t) for t in people]

S = Solution()

#input
flowers = [[1,6],[3,7],[9,12],[4,13]]
people = [2,3,7,11]

print(S.fullBloomFlowers(flowers, people))

File: Solutions/2251. Number of Flowers in Full Bloom/CPP/Number_of_Flowers in_Full_Bloom.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> fullBloomFlowers(vector<vector<int>>& flowers, vector<int>& people) {
        const int flowers_size = flowers.size();

        vector<int> start(flowers_size, 0);
        vector<int> end(flowers_size, 0);

        for (int i = 0; i < flowers_size; ++i) {
            start[i] = flowers[i][0];
            end[i] = flowers[i][1];
        }

        sort(start.begin(), start.end());
        sort(end.begin(), end.end());

        vector<int> result(people.size());
        transform(people.begin(), people.end(), result.begin(), [&](int n) {
            return (upper_bound(start.begin(), start.end(), n) - start.begin()) - (lower_bound(end.begin(), end.end(), n) - end.begin());
        });

        return result;
    }
};

int main() {
    Solution S;

    // input
    vector<vector<int>> flowers = {{1, 6}, {3, 7}, {9, 12}, {4, 13}};
    vector<int> people = {2, 3, 7, 11};

    vector<int> result = S.fullBloomFlowers(flowers, people);
    for (auto& i : result) {
        cout << i << " ";
    }
    cout << endl;

    return 0;
}
